### DISCUSSION TITLE
restart capability

### DISCUSSION THREAD

**Discussion by Ansgar66** [arendst/Tasmota#12306] [Q&A] [answered]
Dear Theo,
first I want to thank you very much for your ideal usage of all these ESP32-applications!
My Idea might be existing already, since it is very simple and nearby:
in case computer hardware connect to the Tasmota hangs, a "restart-capability" would be helpful to trigger a hardware-reset:
the relay stays off only for a time that might be kept in a separate parameter...

best regards
Ansgar

**Comment by Jason2866** [comment, marked as answer]
Which hardware? How do you detect it "hangs"? A specific use case you have in mind would help.
Maybe it is already possible.

**Comment by Ansgar66** [comment]
To go into details:

In my case a raspberry, the switch and the internet router including the VPN-Server are all connected to one Sonoff-PowR2. In case the raspberry hangs I could try a hard reset via switching off and on the Sonoff. But as soon it is switched off I loose access to the Sonoff to switch on again. 

So my wish would be that I could trigger a temporary switch off (off – 10s wait – switch on again). I know this feature e.g. from ePower-Switches…

 

 

 

 

Von: Jason2866 ***@***.***> 
Gesendet: Sonntag, 6. Juni 2021 14:53
An: arendst/Tasmota ***@***.***>
Cc: Ansgar66 ***@***.***>; Author ***@***.***>
Betreff: Re: [arendst/Tasmota] restart capability (#12306)

 

Which hardware? How do you detect it "hangs"? A specific use case you have in mind would help.
Maybe it is already possible.

**Reply by barbudor** [reply]
It's not difficult to achieve.
By setting `PowerOnState 5` and `PulseTime 50`, a `Power OFF` will cut power for 5 seconds and automatically power back on.
You can use rules with `Ping` or `MQTT#disconnected` triggers to decide when powering down.

**Comment by Ansgar66** [comment]
Perfect! Tried and worked!

I knew there was a solution already. Thanks a lot!



Ansgar
